source/extensions/common/wasm/context.cc

```diff
@@ -1233,34 +1233,32 @@ WasmResult Context::defineMetric(uint32_t metric_type, std::string_view name,
   if (metric_type > static_cast<uint32_t>(MetricType::Max)) {
     return WasmResult::BadArgument;
   }
-  auto type = static_cast<MetricType>(metric_type);
-  // TODO: Consider rethinking the scoping policy as it does not help in this case.
-  Stats::StatNameManagedStorage storage(std::string(name), wasm()->scope_->symbolTable());
-  Stats::StatName stat_name = storage.statName();
+  const auto type = static_cast<MetricType>(metric_type);
+  const Stats::DynamicName dynamic_name = Stats::DynamicName(toAbslStringView(name));
   // We prefix the given name with custom_stat_name_ so that these user-defined
   // custom metrics can be distinguished from native Envoy metrics.
   if (type == MetricType::Counter) {
     auto id = wasm()->nextCounterMetricId();
-    auto c = &Stats::Utility::counterFromElements(*wasm()->scope_,
-                                                  {wasm()->custom_stat_namespace_, stat_name});
+    Stats::Counter* c = &Stats::Utility::counterFromElements(
+        *wasm()->scope_, {wasm()->custom_stat_namespace_, dynamic_name});
```

Comment:
s/dynamic_name/name/ here and below.

Reply:
looks like we can't - name (std::string_view) and toAbslStringView(name) both cannot be directly passed as Stats::Element.. am I missing something?

Reply:
and that's why I have DynamicName(...)

Reply:
Well maybe this is due to name is not const ..

Reply:
You are absolutely right. My mistake.

Let's go back to the previous state where you were using StatNameManagedStorage. Sorry about that.